tasks/main.yml

```diff
@@ -37,3 +37,9 @@
   template:
     src: bastion-inventory.j2
     dest: "{{ role_path }}/../../inventories/{{ heat_stack_name }}-autogen-inventory"
+
+- name: Wait for SSH on the bastion node to become available
+  wait_for:
+    host: "{{ hostvars[groups['bastion'][0]]['ansible_ssh_host'] }}"
+    port: 22
+    state: started
```

Comment:
Having the port open does not actually guarantee that the user is able to log in (cloud init may have not yet put the authorized key in place for example). You could maybe copy this:

https://github.com/CSCfi/pouta-ansible-cluster/blob/2fdb6c8f96d153ce63ded36a64c07e3b9439c8f9/playbooks/common/wait_for_ssh.yml